js/symbol_utils.js

```diff
@@ -0,0 +1,112 @@
+// functions related to the symbol groups
+
+// each pie is a group of pie slices and a pie tin (a circle). here add the pie groups
+function addPies() {
+  map.selectAll('g.pie')
+    .data(pieFormData)
+    .enter()
+    .append('g')
+      .classed("pie", true);
+}
+
+function updatePies(category, prevCategory) {
+  
+  // find the pies (groups of slices and tins=circles)
+  var pies = map.selectAll('g.pie');
+  
+  var t1, t2, t3;
+  
+  var oldRadius, newRadius;
+  if(prevCategory === 'piechart') {
+    oldCat = 'total';
+  } else {
+    oldCat = prevCategory;
+  }
+  if(category === 'piechart') {
+    newCat = 'total';
+  } else {
+    newCat = category;
+  }
+  
+  // phase 1: sort and then transition width to 0, height to halfway to new value
+  t1 = pies
+    .sort(function(a,b) { 
+      return d3.descending(a.properties[[category]], b.properties[[category]]);
+    })
+    .transition().duration(500)
+    .attr("transform", function(d) {
+      var halfheight = (scaleCircles(d.properties[[oldCat]]) + scaleCircles(d.properties[[newCat]])) / 2;
+      return "translate(" + d.coordinates.x + "," + d.coordinates.y + ")"+
```

Comment:
if I read this correctly, this `translate` is being applied to each pie/tin group twice for each transition. This is what gets the dot in the right position. I think this is necessary to do each time because some pie/tins are added or removed at each category transition, but I am not sure. 

If we keep the same number of total county pie/tin groups (i.e, we don't drop the ones that are zero in the category), can we avoid running this translate each time and only use it once (when setting up the page)? I know the `scale` is necessary each time, but would reducing/removing the `translate()` calls during transitions help us with performance?

Reply:
Another thought. We ended up switching from js transitions to css transitions for water-use ( [code here](https://github.com/USGS-VIZLAB/water-use/blob/e3f0c454787f69bd56184ab55cd01dafd941acea/layout/js/map.js) ). We used scale3d() after walker looked into different options. Not sure if this is moot now that we are using d3, but css transforms may be a quick option to try (using style instead of transform attrs).

Reply:
We're not currently adding or removing pie elements ever. My idea (not yet implemented) was to hide the unused ones (`display: false`?) but keep everything in place once it's created. I don't actually know whether that's faster or slower than fully removing and re-adding elements at each transition, but my suspicion is it's faster to keep them in place. That's another thing we could experiment with.

Interesting thought to separate the `translate` from the `scale` and only apply the `translate` bit once. I was imagining that any `translate` attribute led to a recalculation each time something was repainted anyway, but again, I suppose I don't actually know.

Notes to myself / other noobs regarding css transitions, which I've never used:
* in the linked code, these lines https://github.com/USGS-VIZLAB/water-use/blob/e3f0c454787f69bd56184ab55cd01dafd941acea/layout/js/map.js#L82-L87 and this line https://github.com/USGS-VIZLAB/water-use/blob/e3f0c454787f69bd56184ab55cd01dafd941acea/layout/js/map.js#L97 seem important.
* here's a concise comparison of d3 vs css transitions: https://stackoverflow.com/questions/46675499/animating-using-d3-transitions-vs-using-css-animation.
* you can chain css transitions: https://css-tricks.com/using-multi-step-animations-transitions/

The existing d3 transitions in this PR definitely need help, not just for speed but because they're not yet doing things in the right order. I think the middle phase is failing to take its place in a chain of transitions and is instead happening ASAP. we could try delays, or try to learn more about how to chain transitions at one level (g.pie) with transitions at a lower level (.pieslice, .tin) and back to transitions at the higher level (g.pie).

Reply:
Cool - yes, those were the lines I was thinking of (was on mobile when I posted and it doesn't seem to let you link to line numbers when in mobile). 

I think you want `display: none;` for getting rid of pies (marty?)

Another advantage _I think_ CSS transitions give us is their handling of being interrupted. Again, I don't know if D3 is smarter than the way we used to use js to animate transitions, so maybe that point is moot. I am referring to the case where you mouse over one category and then another before the transition is complete.

Reply:
D3 is smart about transition interruptions in the sense that if the user initiates a new transition, the old one gets abandoned. I suspect that's the desired behavior, but we should look at that specifically to make sure we're getting what we want with whatever implementation we land on.

Reply:
I did not know you could bake transitions directly into CSS. The reason we've been applying fill styles via D3 is so that we can fade colors in and out, versus changing the class which you can't slow down. Good trick to be able to stick transitions directly into CSS land.

Reply:
Based on that Stackoverflow comment from Alison's link, it looks like simple transitions in CSS will improve performance, but more complex ones are hard to create in just CSS so you might need to stick to D3.

Reply:
true. however, coin flips ARE pretty simple transitions. i think it's probably possible, unfamiliar though it is to the two of us.

Reply:
Right - we should most certainly try out transitioning via CSS for coin flips! I wasn't trying to imply this one was hard, but was confirming the use cases for d3 to transition vs css.

Reply:
oh, yes, agreed! for example, i doubt css could do wiper transitions, which involve recalculating the contents of the `d` path element string. (then again, i still know next to nothing about css transitions, so...)

ps the wiper transitions were looking pretty cool except for being slow. we should totally try them in some future viz.